internal/xcuitest/config.go

```diff
@@ -226,3 +230,58 @@ func SortByHistory(suites []Suite, history insights.JobHistory) []Suite {
 	}
 	return res
 }
+
+// ShardSuites applies sharding by provided shard config.
+func ShardSuites(p *Project) error {
+	var suites []Suite
+	for _, s := range p.Suites {
+		if s.Shard != "concurrency" {
+			suites = append(suites, s)
+			continue
+		}
+		shardedSuites, err := parseTestClassesFile(s, p.Sauce.Concurrency)
+		if err != nil {
+			log.Warn().Err(err).Msgf("Failed to get tests from %q. Using default settings.", s.TestClassesFile)
+			suites = append(suites, s)
+			continue
+		}
+		if len(shardedSuites) == 0 {
+			suites = append(suites, s)
+		} else {
+			suites = append(suites, shardedSuites...)
+		}
+	}
+	p.Suites = suites
+
+	return nil
+}
+
+func parseTestClassesFile(suite Suite, ccy int) ([]Suite, error) {
```

Comment:
After doing a second pass on it, I think we need a better name or a set of methods to improve the clarity of intent.

You call it `parse test classes file`, but it's doing way more than that.

And when you read it from a caller's perspective, you see:
```go
shardedSuites, err := parseTestClassesFile(s, p.Sauce.Concurrency)
```

Which makes you wonder why you got sharded suites back, when you only asked it to parse a test classes file?